pkg/doc/signature/ed25519signature2018/suite_crypto_test.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ed25519signature2018
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/google/tink/go/keyset"
+
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto"
+	sigverifier "github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier"
+	kmsapi "github.com/hyperledger/aries-framework-go/pkg/kms"
+	"github.com/hyperledger/aries-framework-go/pkg/kms/localkms"
+	mockkms "github.com/hyperledger/aries-framework-go/pkg/mock/kms"
+	"github.com/hyperledger/aries-framework-go/pkg/mock/storage"
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock/noop"
+)
+
+func TestNewCryptoSignerAndVerifier(t *testing.T) {
+	lKMS := createKMS()
+
+	kid, kh := createKeyHandle(lKMS, kmsapi.Ed25519Type)
+	if kid == "" {
+		panic("kid is empty")
+	}
+
+	tinkCrypto, err := tinkcrypto.New()
+	if err != nil {
+		panic("failed to create tinkcrypto")
+	}
+
+	doc := []byte("test doc")
+
+	suiteSigner := NewCryptoSigner(tinkCrypto, kh)
+	suiteVerifier := NewCryptoVerifier(&Crypto{
+		Crypto:   tinkCrypto,
+		localKMS: lKMS,
+	})
+
+	ss := New(WithSigner(suiteSigner), WithVerifier(suiteVerifier))
+
+	docSig, err := ss.Sign(doc)
+	if err != nil {
+		panic("failed to create a signature")
+	}
+
+	pubKH, err := kh.Public()
+	if err != nil {
+		panic("failed to get a public handle of the key")
+	}
+
+	// verify with key handle
+	err = ss.Verify(pubKH, doc, docSig)
+	if err != nil {
+		panic("failed to verify a signature")
+	}
+}
+
+// LocalCrypto defines a verifier which is based on Local KMS and Crypto
+// which uses keyset.Handle as input for verification.
+type Crypto struct {
+	*tinkcrypto.Crypto
+	localKMS *localkms.LocalKMS
+}
+
+func (t *Crypto) Verify(sig, msg []byte, kh interface{}) error {
+	var keyHandle *keyset.Handle
+
+	switch pubKey := kh.(type) {
+	case *keyset.Handle:
```

Comment:
I am not sure in what case handle would be passed in. Normally use case:
1. retrieve key from did (pub key bytes, type) 
2. use pub key for verification.

Reply:
usually pub key bytes will be transformed into public keyset.Handle to use kms/crypto's default implementation services

Reply:
I changed `publicKeyResolver.Resolve()` and `verifier.Verify()` to take `PublicKey` instead of genetic `interface{}`. The reason is that we don't see any use case to take e.g.  `*keyset.Handle` for now.
It looks like that when we would like to use `tinkcypto.Verifier`, we can intercept the call and convert `PublicKey` into the `*keyset.Handle` using `PubKeyBytesToHandle()`. This approach is implemented in **suite_crypto_test.go**.